actions/dispatch-workflow/action.yml

```diff
@@ -0,0 +1,116 @@
+name: Dispatch Workflow
+description: Dispatch a github workflow and return the run id
+
+inputs:
+  repo:
+    description: Owner and repo in the form owner/repo
+    required: true
+  workflow:
+    description: The name of the workflow file
+    required: true
+  ref:
+    description: The ref or branch of the repo to use
+    required: true
+  workflow-dispatch-fields:
+    description: |
+      Fields to pass into the workflow dipatch, example:
+        --field distinct_run_name="${DISTINCT_RUN_NAME}" \
+        --field evm-ref="${EVM_REF}"
+    required: false
+  distinct_run_name:
+    description:
+      The distinct run name added to the workflow, this is required to more
+      accurately find the correct workflow that was started
+    required: true
+  actor:
+    description:
+      The github actor used to dispatch the workflow, required to more
+      accurately find the correct workflow that was started
+    required: true
+  github-token:
+    description: The github token to use
+    required: true
+  retry-delay:
+    description:
+      The amount of time in seconds to sleep between calls to check for workflow
+      to be created
+    required: false
+    default: "10"
+  retry-count:
+    description: The max number of times to wait for the workflow to be created
+    required: false
+    default: "6"
+outputs:
+  run-id:
+    description: The found run-id
+    value: ${{ steps.get-run-id.outputs.run-id }}
+
+runs:
+  using: composite
+  steps:
+    - name: Dispatch Workflow
+      shell: bash
+      id: get-run-id
+      env:
+        GH_TOKEN: ${{ inputs.github-token }}
+        DISTINCT_RUN_NAME: ${{ inputs.distinct_run_name }}
+        GH_REPO: ${{ inputs.repo }}
+        WORKFLOW: ${{ inputs.workflow }}
+        REF: ${{ inputs.ref }}
+        ACTOR: ${{ inputs.actor }}
+        EVM_REF: ${{ inputs.evm-ref }}
+        FIELDS: ${{ inputs.workflow-dispatch-fields }}
+        RETRY_DELAY: ${{ inputs.retry-delay }}
+        RETRY_COUNT: ${{ inputs.retry-count }}
+      run: |
+        current_time=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
+
+        # run the workflow
+        echo "Dispatching the workflow"
+        gh workflow run "${WORKFLOW}" \
+          --repo ${GH_REPO} \
+          --ref ${REF} ${FIELDS}
+
+        # Maximum number of attempts to check for the workflow run
+        max_attempts=$(($RETRY_COUNT + 0))
+        # Time (in seconds) to wait between attempts
+        wait_time=$(($RETRY_DELAY + 0))
+
+        attempt=1
+
+        while [ $attempt -le $max_attempts ]; do
+          echo "Attempt $attempt: Checking for workflow runs..."
+
+          # Fetch workflow runs created after the script start time
+          workflow_runs=$(curl -H "Authorization: token $GH_TOKEN" \
+            -H "Accept: application/vnd.github.v3+json" \
+            "https://api.github.com/repos/${GH_REPO}/actions/runs?actor=$ACTOR&event=workflow_dispatch&exclude_pull_requests=true&created=>=$current_time")
```

Comment:
Is there a reason why you're using a curl and not `gh` CLI?

Reply:
ahh good call out! if we use `gh` cli then we can get rid of all references of `${GH_REPO}` since it automatically reads it.

Reply:
I only ever used the gh cli to dispatch the workflow. The gh cli versions for other calls typically didn't give me all the data as easily or required more complex parameters to pass custom json queries making the code harder to read. I will convert the one usage of gh cli so they are consistent in what they use. I have found the gh cli great for the use cases they expect you to want but leave out some of the goodies you get from the api for free.

Reply:
I guess I can use the `gh api ...` option instead of curl. I will go that route instead

Reply:
Cleaned up in: https://github.com/smartcontractkit/.github/pull/454/commits/e02e6b59110bbf3ca6c14a23d1fa1641d2b0bde0